common/lib/common-utils/src/idleTaskScheduler.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * API used to schedule non-essential tasks
+ * Time measurements are in milliseconds as a floating point with a decimal
+ * Takes in and runs a callback during idle time. Fallback to setTimeout if window doesn't
+ * support requestIdleCallback.
+ * @returns A promise pertaining to the callback that was passed in.
+ */
+export async function scheduleIdleTask<T>(callback: () => T, timeout: number): Promise<T> {
+    // Check for the availability in window.
+    return typeof window?.requestIdleCallback === "function"
```

Comment:
please search Teams (or internet) for a discussion about globalThis - I believe you want to use it instead of "window" object, as I think this code will throw exception in Node.js environment (and likely same - in web worker).